homeassistant/components/roon/media_player.py

```diff
@@ -163,6 +163,76 @@ def update_data(self, player_data=None):
 
     def update_state(self):
         """Update the power state and player state."""
+
+        def _parse_volume(player_data):
+            """Parse volume data to determine volume levels and mute state."""
+            volume = {
+                "level": 0,
+                "step": 0,
+                "muted": False,
+            }
+
+            try:
+                volume_data = player_data["volume"]
+                volume_muted = volume_data["is_muted"]
+                volume_step = convert(volume_data["step"], int, 0)
+
+                if volume_data["type"] == "db":
+                    level = convert(volume_data["value"], float, 0.0) / 80 * 100 + 100
+                else:
+                    level = convert(volume_data["value"], float, 0.0)
+
+                volume_level = convert(level, int, 0) / 100
+            except KeyError:
+                # catch KeyError
+                pass
+            else:
+                volume["muted"] = volume_muted
+                volume["step"] = volume_step
+                volume["level"] = volume_level
+
+            return volume
+
+        def _parse_now_playing(player_data):
+            """Parse now playing data to determine title, artist, position, duration and artwork."""
+            now_playing = {
+                "title": None,
+                "artist": None,
+                "album": None,
+                "position": 0,
+                "duration": 0,
+                "image": None,
+            }
+            now_playing_data = None
+
+            try:
+                now_playing_data = player_data["now_playing"]
+                media_title = now_playing_data["three_line"]["line1"]
+                media_artist = now_playing_data["three_line"]["line2"]
+                media_album_name = now_playing_data["three_line"]["line3"]
+                media_position = convert(now_playing_data["seek_position"], int, 0)
+                media_duration = convert(now_playing_data["length"], int, 0)
+            except KeyError:
+                # catch KeyError
+                pass
+            else:
+                now_playing["title"] = media_title
+                now_playing["artist"] = media_artist
+                now_playing["album"] = media_album_name
+                now_playing["position"] = media_position
+                now_playing["duration"] = media_duration
+
+                try:
```

Comment:
```suggestion
                if image_id:
                    now_playing["image"] = self._server.roonapi.get_image(image_id)
                try:
```

Reply:
simplifies the double try catch. no longer need to try since get will gracdfully set to None if no key exists.